File: 1996. The Number of Weak Characters in the Game.py
class Solution:

    def numberOfWeakCharacters(self, properties: List[List[int]]) -> int:
        properties = sorted(properties, key=lambda row: (-row[0], row[1]))
        
        ans, maxTillNow = 0, float('-inf')
        for _, p in properties:
            if maxTillNow > p:
                ans += 1
            else:
                maxTillNow = p
                
        return ans

File: 1996. The Number of Weak Characters in the Game.cpp
class Solution {
public:
    static bool sagarhasan273(vector<int> &a, vector<int> &b){
        if (a[0] != b[0]){
            return a[0]>b[0];
        }return a[1]<b[1];
    }
    
    int numberOfWeakCharacters(vector<vector<int>>& properties) {
        int maxTillNow = INT_MIN, ans=0;
        
        sort(properties.begin(), properties.end(), sagarhasan273);
        
        for (auto p: properties){
            if (maxTillNow > p[1]) ans++;
            else maxTillNow = p[1];
        }
        
        return ans;
    }
};